### PR TITLE
Create Groups

### LINKED ISSUE #138: Build the "Create Group" Page
We need a page that contains a form allowing users to create a group. This should be accessible through the group list page (when there are no groups and when there are groups).

### PR DESCRIPTION
**Pull request by Waidhoferj**
Fixes #138
Users can now create groups on the Group List page!

## Testing
1. Go to the Group List Page
2. Click on the plus icon in the upper right
3. Fill out the fields and press "Create Group"
4. Verify that you get a toast with an explanation and the group appears in the list.

### CODE REVIEW

File: frontend/src/components/Popup/Popup.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import "./Popup.scss";
+import { motion } from "framer-motion";
+interface PopupProps {
+  children?: React.ReactNode;
+  onClose: () => void;
+}
+
+export default function Popup(props: PopupProps) {
```
Comment (gwholland3) [inline frontend/src/components/Popup/Popup.tsx:8]: Is this something we can insert anywhere?

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx [hunk 2]
```diff
@@ -94,8 +94,7 @@ function GroupHub({ group }: GroupHubProps) {
 
     if (resp["data"]["leaveGroup"]["success"]) {
       push("/groups");
-    }
-    else {
+    } else {
```
Comment (gwholland3) [inline frontend/src/pages/Groups/GroupPage/GroupPage.tsx:97]: Was this and other formatting stuff suggested by Prettier or something?

**Review by gwholland3 — APPROVED** [review]
LGTM!

**Comment by gwholland3** [comment]
Actually one thing I just noticed: when creating the group, it seems like the description text requires a minimum of 10 characters, but you don't get an error message telling you that if you try to submit with fewer

**Comment by gwholland3** [comment]
And one last bug I noticed: If you leave a group that you were the last member of, it appears as though the group is deleted, both on your groups page and in the graphql interface. However, if you had copied the url of that page and try to visit it after it's deleted, you get an error instead of the "couldn't find the group you are looking for" message.

Not sure if this is related to your changes or if this should be opened as a separate issue

**Comment by Waidhoferj** [comment]
@gwholland3 I think the group getting deleted should be expected behavior. If no one belongs to that group, it wouldn't be reachable and we would be storing data for no reason.

I addressed the length issue and added more descriptive error handling. I don't think we need to specify length so just took it out.

File: frontend/src/components/Popup/Popup.tsx (hunk 1 above)
Comment (Waidhoferj) [reply frontend/src/components/Popup/Popup.tsx:8]: Yeah! The popup component should work on any screen. Just wrap it in an `AnimatePresence` and you'll get the animations too.

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx (hunk 2 above)
Comment (Waidhoferj) [reply frontend/src/pages/Groups/GroupPage/GroupPage.tsx:97]: Yes

**Comment by gwholland3** [comment]
@Waidhoferj yeah I understand that the group should be deleted when there is no one in it, but what I though was possibly a bug is that if I try to visit `/group/any_random_id_string`, it gives me a nicely formatted screen telling me that it couldn't find said group, with a go back button. 

However, if I try to visit the url of a group that was just deleted (by me being the only member and leaving it), then I get an ugly react error.

**Comment by Waidhoferj** [comment]
I also changed the catch all route from a redirect back to a normal route component. If we use the redirect component, the autologin functionality wont work because in the split second before the user is auto logged in, they are redirected.

**Review by gwholland3 — APPROVED** [review]
Looks good! Nice fixes

**Comment by kyleshare** [comment]
Looks good! Only problem I noticed is that the "Artworks" metric is not being updated from 0 when I add an artwork to the group I created.

**Comment by Braden50** [comment]
> Looks good! Only problem I noticed is that the "Artworks" metric is not being updated from 0 when I add an artwork to the group I created.

I have an issue for this (that is assigned to me). I'm not sure if I will get it fixed in time, but I will look into it.

**Review by Braden50 — APPROVED** [review]
Looks beautifully simple on the front end. Nice work! Scroll bars are still many places in the app including the pop up here, but that is totally ok given our timeline.

**Pull request merged**